guardian/managers.py

```diff
@@ -67,14 +81,43 @@ def bulk_assign_perm(self, perm, user_or_group, queryset):
         assigned_perms = []
         for instance in queryset:
             if not checker.has_perm(permission.codename, instance):
-                kwargs = {'permission': permission, self.user_or_group_field: user_or_group}
-                if self.is_generic():
-                    kwargs['content_type'] = ctype
-                    kwargs['object_pk'] = instance.pk
-                else:
-                    kwargs['content_object'] = instance
+                kwargs = self._perm_kwargs(permission, user_or_group, instance, ctype, user_group_object)
                 assigned_perms.append(self.model(**kwargs))
-        self.model.objects.bulk_create(assigned_perms)
+        self.bulk_create(assigned_perms)
```

Comment:
I do not remember. Perhaps it was not working or perhaps I noticed that it was referring to itself so I simplified it.